### PR TITLE
Use camtrapdp to read pkgs

### PR DESCRIPTION
**Pull request by damianooldoni**
This PR is the first step towards version 1.0. 

**IMPORTANT**: this PR doesn't point to `main` but to dev-branch `version-1.0` which will be the reference branch for developing camtraptor v1.0. As I didn't modify any other function, the vast majority of the tests will not work. I can deal with it at the moment as we do the changes stepwise via multiple PRs.

This PR focuses on the reading Camtrap DPs functionality. This PR does the following:

- Read packages by reexporting `camtrapdp::read_camtrapdp()` (see #298).
- Deprecate `read_camtrap_dp()`.
- Remove check functions as data package checks are now done using `camtrapdp::check_camtrapdp()`. Remove the section about validation functions in `_pkgdown.yml` as well.
- Remove all help functions used to downconvert to 0.1.6.
- Remove help functions to add taxonomic information to `observations` data.frame as it is done by camtrapdp internally.
- Remove `write_dwc.R` as this functionality is developed in camtrapdp, see https://github.com/inbo/camtrapdp/pull/55
- Remove `write_eml.R`: not needed anymore, see h``ttps://github.com/inbo/camtrapdp/issues/61
- Rename `zzz.R` to `utils.R`.
- Remove tests of removed functions. 
- Adapt tests related to reading camtrap DPs. These are very minimal now as the most of the tests are within camtrapdp. We test now only that the `read_camtrap_dp()` returns the exact warning and that its output is identical to the output of `camtrapdp:: read_camtrapdp()`.
- Update `README` and `NEWS.md`.
- Add `example_dataset` function. Add memoise::memoise()`as camtrapdp's `*_cache()` function is not exported. See #312

### CODE REVIEW

**Comment by peterdesmet** [comment]
Have not reviewed files, just read the PR description. Regarding this:

> Adapt tests related to reading camtrap DPs. These are very minimal now as the most of the tests are within camtrapdp. We test now only that the read_camtrap_dp() returns the exact warning and that its output is identical to the output of camtrapdp::read_camtrapdp().

I would only test that the output is the same, not the warnings. Alternatively, test _only_ that a deprecation warning is returned, pointing to `read_camtrapdp()`

**Review by peterdesmet — CHANGES_REQUESTED** [review]

File: _pkgdown.yml [hunk 1]
```diff
@@ -16,9 +16,6 @@ authors:
     html: "<img src='https://www.lifewatch.be/sites/lifewatch.be/themes/lifewatchportal/images/header_lw_logo.png' width=72>"
 
 reference:
-- title: Read data
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply _pkgdown.yml:19]: Good point. This is one thing left by my first refactoring when I thought to not include the reading functionality in camtraptor. I will add it.

File: tests/testthat/test-read_camtrapdp.R [hunk 2]
```diff
@@ -0,0 +1,25 @@
+test_that("read_camtrapdp() reads a Camtrap DP", {
+  skip_if_offline()
+  file <-
+    "https://raw.githubusercontent.com/tdwg/camtrap-dp/1.0/example/datapackage.json"
+  expect_identical(read_camtrapdp(file), camtrapdp::read_camtrapdp(file))
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply tests/testthat/test-read_camtrapdp.R:5]: Well, I have to test that the output of camtraptor's function is the same as the output of the camtrapdp's function, i.e. the reexport has been done correctly. So, I expect that the output of the two functions is exactly the same (identical).

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# camtraptor 1.0
+
+- `read_camtrapdp()` is the new function to read Camera Trap DPs. It is reexported from camtrapdp's function [`read_camtrapdp()`](https://inbo.github.io/camtrapdp/reference/read_camtrapdp.html) (#298).
+- `read_camtrap_dp()` is deprecated. Use `read_camtrapdp()` instead.
+- publication functions `write_dwc()` and `write_eml()` are removed. The function `write_dwc()` has been moved to camtrapdp package. The `write_eml()` function is not needed anymore ([camtrapdp(#61)](https://github.com/inbo/camtrapdp/issues/61)).
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply NEWS.md:5]: Nice rephrasing 🔝

File: tests/testthat/test-read_camtrapdp.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (damianooldoni) [reply tests/testthat/test-read_camtrapdp.R:5]: If my answer makes sense, feel free to resolve this conversation.

[the remark this answers is not in the document]
Comment (peterdesmet) [reply tests/testthat/test-read_camtrapdp.R:5]: Oh, this test file is a mix of `read_camtrapdp()` and `read_camtrap_dp()`. Personally, I would not test any of the reexported functions.

File: DESCRIPTION [hunk 4]
```diff
@@ -34,6 +34,7 @@ Depends:
     R (>= 3.5.0)
 Imports:
     assertthat,
+    camtrapdp,
     dplyr (>= 1.1.0),
     EML,
     frictionless,
```
Comment (peterdesmet) [inline DESCRIPTION:40]: I wonder if `frictionless` this needs to be imported still?

**Review by peterdesmet — CHANGES_REQUESTED** [review]

File: R/get_cam_op.R [hunk 5]
```diff
@@ -15,7 +15,7 @@
 #' https://jniedballa.github.io/camtrapR/reference/cameraOperation.html).
 #'
 #' @param package Camera trap data package object, as returned by
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/get_cam_op.R:17]: I agree. I would not deprecated `package` argument even. The typical user doesn't pass this argument by name as `get_*(package = )`.

[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/get_cam_op.R:17]: I think this should be tackled in apart PR. I wrote an issue about it: https://github.com/inbo/camtraptor/issues/311

File: R/check_species.R [hunk 6]
```diff
@@ -4,11 +4,10 @@
 #' metadata (`package$taxonomic`) and returns error if not.
 #'
 #' @param package Camera trap data package object, as returned by
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/check_species.R]: Solved in [87460ac](https://github.com/inbo/camtraptor/pull/307/commits/87460ac6cb74a4374948ac9b713dec9af0fde653)

File: R/read_camtrap_dp.R [hunk 7]
```diff
@@ -1,163 +1,16 @@
...
+#' Reads files from a [Camera Trap Data Package (Camtrap DP)](
 #' https://camtrap-dp.tdwg.org) into memory.
-#' All datetime information is automatically transformed to Coordinated
-#' Universal Time (UTC).
-#' Vernacular names found in the metadata (`package$taxonomic`) are added to the
-#' `observations` data frame.
+#' 
+#' This function is deprecated. Please, use `read_camtrapdp()` instead.
 #'
 #' @param file Path or URL to a `datapackage.json` file.
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/read_camtrap_dp.R]: Yes, indeed, we can do it. Done in [20c782b](https://github.com/inbo/camtraptor/pull/307/commits/20c782b0416d7feddd9f0e981ca0f577599d1676). The syntax to use is:
`#' @inheritParams packageName::functionName`

File: _pkgdown.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (damianooldoni) [reply _pkgdown.yml]: Done in [fb5235e](https://github.com/inbo/camtraptor/pull/307/commits/fb5235edc8972a67bdc421d6752c665b3930dd2b)

File: DESCRIPTION (hunk 4 above)
Comment (damianooldoni) [reply DESCRIPTION]: Indeed, not needed anymore. Removed in [064a7e3].(https://github.com/inbo/camtraptor/pull/307/commits/064a7e3e7ce9a94f0efd1b8abe36e34ce1ab8f48)

File: R/camtraptor-package.R [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#' @keywords internal
+"_PACKAGE"
+
+## usethis namespace: start
+#' @importFrom lifecycle deprecated
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/camtraptor-package.R]: Indeed. Not needed! Removed in 93c1abd.

File: R/camtrapdp_functions.R
[the remark this answers is not in the document]
Comment (damianooldoni) [reply R/camtrapdp_functions.R:1]: It makes sense. Done in 2561536.

File: man/figures/lifecycle-archived.svg
[the remark this answers is not in the document]
Comment (damianooldoni) [reply man/figures/lifecycle-archived.svg:1]: Unfortunately not. From [badge documentation](https://lifecycle.r-lib.org/reference/badge.html):

> Call [usethis::use_lifecycle()](https://usethis.r-lib.org/reference/use_lifecycle.html) to copy the badge images into the man/ folder of your package.

Notice that dplyr and purrr R packages do the same, see svg files in `/man/figures` folder: 
- [dplyr](https://github.com/tidyverse/dplyr/tree/main/man/figures)
- [purrr](https://github.com/tidyverse/purrr/tree/main/man/figures)

**Review by peterdesmet — APPROVED** [review]

**Review by PietrH — APPROVED** [review]

**Comment by PietrH** [comment]
I made some minor suggestions, lots of test failures.

File: NEWS.md [hunk 9]
```diff
@@ -1,3 +1,11 @@
+# camtraptor 1.0
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply NEWS.md:1]: Done in [da617fc](https://github.com/inbo/camtraptor/pull/307/commits/da617fcb9c38b235a7297d3a6d4e0ff25efe9ed3)

File: NEWS.md [hunk 10]
```diff
@@ -1,3 +1,11 @@
...
+
+- `read_camtrapdp()` reexports the read functionality from [camtrapdp](https://inbo.github.io/camtrapdp/reference/read_camtrapdp.html) (#298). 
+It replaces the now deprecated `read_camtrap_dp()` (with underscore) (#298).
+- `read_camtrap_dp()` is deprecated. Use `read_camtrapdp()` instead.
+- `write_dwc()` has moved to [camtrapdp](https://inbo.github.io/camtrapdp/reference/write_dwc.html). `write_eml()` is not needed for GBIF processing of Camtrap DPs and has been removed ([camtrapdp#61](https://github.com/inbo/camtrapdp/issues/61)).
+- `example_dataset()` allows to create an example dataset in Camtrap DP 1.0 format. It reexports the functionality from [camtrapdp](https://inbo.github.io/inbo/camtrapdp/reference/example_dataset.html). The internal dataset `mica` is not present anymore.
+
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply NEWS.md:11]: Done in https://github.com/inbo/camtraptor/commit/da617fcb9c38b235a7297d3a6d4e0ff25efe9ed3.

File: NAMESPACE [hunk 11]
```diff
@@ -28,11 +29,13 @@ export(pred_notin)
...
+export(read_camtrapdp)
 export(read_wi)
 export(round_coordinates)
-export(write_dwc)
-export(write_eml)
+importFrom(camtrapdp,example_dataset)
+importFrom(camtrapdp,read_camtrapdp)
 importFrom(dplyr,"%>%")
 importFrom(dplyr,.data)
+importFrom(memoise,memoise)
```
[the remark this answers is not in the document]
Comment (damianooldoni) [reply NAMESPACE]: Good catch! No, indeed. Removed in 506e8e6.

**Comment by damianooldoni** [comment]
@PietrH: indeed, we expect at this stage a lot of test failures as we are just started the transition.

But the important stuff is that the following tests are not failing:

```r
> devtools::test(filter = "read_camtrapdp")
ℹ Testing camtraptor
✔ | F W  S  OK | Context
✔ |          2 | read_camtrapdp [2.0s]                                                                     

══ Results ════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 2.1 s

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 2 ]
```

**Pull request merged**